crates/superposition_types/Cargo.toml

```diff
@@ -8,23 +8,34 @@ edition = "2021"
 [dependencies]
 # env
 actix-web = { workspace = true, optional = true }
-log = { workspace = true }
-serde = { workspace = true }
-serde_json = { workspace = true }
+anyhow = { workspace = true, optional = true }
+cac_db_config = { path = "../cac_db_config", optional = true }
+chrono = { workspace = true }
 derive_more = { workspace = true }
-thiserror = { version = "1.0.57", optional = true }
 diesel = { workspace = true, optional = true }
-anyhow = { workspace = true, optional = true }
+experimentation_db_config = { path = "../experimentation_db_config", optional = true }
 jsonlogic = { workspace = true }
+log = { workspace = true }
 regex = { workspace = true }
+serde = { workspace = true }
+serde_json = { workspace = true }
 superposition_derives = { path = "../superposition_derives", optional = true }
+thiserror = { version = "1.0.57", optional = true }
+uuid = { workspace = true }
 
 [features]
 default = ["server"]
-diesel_derives = ["dep:diesel", "dep:superposition_derives"]
+diesel_derives = [
+    "dep:diesel",
+    "dep:superposition_derives",
+    "dep:cac_db_config",
+    "cac_db_config/diesel_derives",
+    "experimentation_db_config?/diesel_derives",
```

Comment:
What does `?` do here?

Reply:
it is used while enabling certain feature of another dependency which in turn in optional and is dependent on some other feature to enable the usage of that dependency
check the section before the second Note [here](https://doc.rust-lang.org/cargo/reference/features.html#dependency-features)